contents/testing/security-testing.md

```diff
@@ -0,0 +1,91 @@
+
+The goal of security testing is to:
+- Identify the threats in the system.
+- To measure the potential vulnerabilities of the system.
+- To help in detecting every possible security risks in the system.
+- To help developers in fixing the security problems through coding.
+
+:fas-info-circle: Knowledge of Software Development Life-Cycles (SDLCs) would be useful but not necessary.
+
+![Generic SDLC Model](security-testing/Sec-Test-Fig-1.png "Generic SDLC Model")
+
+## Why Have Security Testing?
+
+A comprehensive security testing framework deals with validation across all layers of an application. Through this, the organization can evaluate their application code for vulnerabilities and take remedial measures for the same. Recently, many of the software development organizations have been making use of secure software development life cycle methodologies to ensure identification and rectification of vulnerability areas early on in the application development process.
+
+__Security Testing helps to achieve__:
+- Confidentiality of Information.
+- Integrity of Information.
+- Authenticity of Identity.
+- Authorisation: The process of determining that a requester is allowed to receive a service or perform an operation (e.g. Access Control).
+- Availability: Assuring information and communications services will be ready for use when expected.
+- Non-Repudiation: In reference to digital security, non-repudiation means to ensure that a transferred message has been sent and received by the parties claiming to have sent and received the message.
+
+## How to Get Started with Security Testing?
+
+As a subset of software testing, security testing operates in tandem with your SDLC. It is a conscious decision that needs to be made for effective application.
+
+![OWASP Testing Framework Workflow](security-testing/Sec-Test-Fig-2.png "OWASP Testing Framework Workflow")
+
+### Phase 1: Before Development Begins
+
+Define an SDLC and review policy standards. Ensure that there are appropriate policies, standards, and documentation in place. Documentation is extremely important as it gives development teams guidelines and policies that they can follow. **People can only do the right thing if they know what the right thing is.**
+
+### Phase 2: During Definition and Design
+
+Review security requirements, always make sure you have a security policy! Security requirements define how an application works from a security perspective. It is essential that the security requirements are tested. Testing in this case means testing the assumptions that are made in the requirements and testing to see if there are gaps in the requirements definitions.
```

Comment:
Perhaps can provide more info (e.g. in terms of a tool tip) on security policy as it's was not referred in the article?

Reply:
Do you happen to know how I can add a tooltip? Now that I think about it I could probably add one to the introduction of a SDLC definition earlier haha

Reply:
```
<tooltip content="Description when hovered">
 _security policy_
 </tooltip>
```

Reply:
Thank you! :)